classes/class-admin.php

```diff
@@ -728,19 +759,165 @@ public function wp_ajax_reset() {
 	private function erase_stream_records() {
 		global $wpdb;
 
-		$where = '';
+		// If this is a multisite and it's not networked activated,
+		// only delete the entries from the blog which made the request.
+		if ( self::WP_STREAM_MULTI_NOT_NETWORK === $this->plugin->get_site_type() ) {
```

Comment:
<!-- potential concern, ask for clarification or investigation -->
**❓question:** I guess my only concern here is that the user may still want to erase Stream data of a single site even if Stream is network-activated.

The reason is that in order to trigger the data removal, you need to go to a specific site -> Stream -> Settings and click that button there. It may be misleading (even if we have the warning message) that triggering the removal on that particular site will affect the entire network. Also, a "regular" admin on a child site may be able to initiate a destructive action for the entire network (which only super-admins should be able to do).

I think we may need to make that action more specific and by default only remove single site data, even if Stream is network activated. On top of that, we would either need:
- a network-wide Stream Settings page where the entire network data removal would be possible, or
- a second button on the current Settings page which would delete data from the entire network and would only be displayed for super-admins.

Are my concerns valid?

Reply:
@delawski 

> I guess my only concern here is that the user may still want to erase Stream data of a single site even if Stream is network-activated.

This would be adding a feature which is not currently there. We could do that but we would need some more checks to make sure we're not trying to delete the same thing more than once? I'd kind of prefer to put this in either the next release or in a separate issue for ease of backing out if it needs to be.

> The reason is that in order to trigger the data removal, you need to go to a specific site -> Stream -> Settings and click that button there.

When the plugin is network activated, the only place to remove data is in the network admin. The button to remove data isn't in the individual sites. And vice versa -- when the plugin is not network activated, it doesn't show up in the network admin. 

On your local environment, the `is_network_activated()` check doesn't work correctly due to the rewriting of `stream` to `stream-src`, I need to make an issue for that but keep forgetting. I have a mu plugin with that filter which I use to test it being network activated:
`add_filter( 'wp_stream_is_network_activated', '__return_true' );`

I should probably write an e2e test for the two multisite possibilities? Although that would require fixing that rewrite issue.

Reply:
> I'd kind of prefer to put this in either the next release or in a separate issue for ease of backing out if it needs to be.
💯 agree on that - let's not work on it in this release.

Also, in light of what you just explained to me:

> When the plugin is network activated, the only place to remove data is in the network admin. The button to remove data isn't in the individual sites. And vice versa -- when the plugin is not network activated, it doesn't show up in the network admin.

it may not be necessary to add such a feature at all. The user flows are correct and messaging is clear 👍